openssl-dynamic/src/main/c/ssl.c

```diff
@@ -2118,16 +2245,24 @@ TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterial)(TCN_STDARGS, jlong ssl, jlong chai
 
     TCN_CHECK_NULL(cchain, chain, /* void */);
 
-    numCerts = sk_X509_num(cchain);
+#ifdef OPENSSL_IS_BORINGSSL
+    for (i = 0; i < numCerts; i++) {
+        certs[i] = sk_CRYPTO_BUFFER_value(cchain, i);
+    }
 
+    if (numCerts <= 0 || SSL_set_chain_and_key(ssl_, certs, numCerts, pkey, NULL) <= 0) {
```

Comment:
Once https://github.com/netty/netty-tcnative/pull/446 is merged we need to ensure to pass in the "private key method" if pkey is `NULL` here